### ISSUE TITLE
Need more examples of using `main_binary` and `cargo_binary`.

### ISSUE THREAD

**Issue by gnzlbg** [killercup/assert_cli#89]
The docs don't show how to test a cli application written in Rust. 

Like, there are no examples in the documentation that actually show how to do this. The `main_binary` and `cargo_binary` functions are mentioned, but without showing how to use them to do anything useful.

If this crate main audience are Rust programmers wanting to test Rust CLI applications, having some examples showing how to test an actual Rust cli application would significantly lower the barrier of entry to use this crate.

**Comment by killercup** [comment]
Agreed, these functions need docs. I’ve been meaning to get back to this
crate and the ongoing API refactor (I’m sorry I haven’t been around much,
@epage!) and more detailed docs are most certainly on the list!

For now, though, do you have any concrete issues with these helpers? (The
are just shortcuts for `cargo run --`.)

gnzlbg <notifications@github.com> schrieb am So. 18. Feb. 2018 um 11:29:
